alembic/env.py

```diff
@@ -20,8 +21,8 @@
 #from flask import current_app
 #config.set_main_option('sqlalchemy.url', current_app.config.get('SQLALCHEMY_BINDS')['libraries'])
 #target_metadata = current_app.extensions['migrate'].db.metadata
-target_metadata = None
-
+#target_metadata = None
+target_metadata = biblib.models.Base.metadata
```

Comment:
I think its based off of whether the models need to be imported for the revisions to work. If the revisions are defined without using the model, or can be redefined without the models without making a mess, then I don't think you need it. `biblib` tends to be the exception when it comes to its `sqlalchemy` implementation...

Reply:
Having said all that. It looks like I managed to refactor the `biblib` dependencies out of the migrations so it isn't actually needed in this case.

Reply:
Digging a little deeper, the `metadata` needs to be set for `autogenerate` to work and create proper revisions, per this comment in `env.py`. 

```python
# add your model's MetaData object here
# for 'autogenerate' support
# from myapp import mymodel
# target_metadata = mymodel.Base.metadata
```

It looks like it might be able to be removed after the autogeneration is complete, but I think that still depends on whether the revisions depend on the models. In this case, it looks like I did refactor the models out of the revisions and the database upgrade does seem to still behave.